jupyter_core/migrate.py

```diff
@@ -198,24 +209,35 @@ def migrate():
         'ipython_dir': get_ipython_dir(),
         'profile': os.path.join(get_ipython_dir(), 'profile_default'),
     }
+    migrated = False
     for src_t, dst_t in migrations.items():
         src = src_t.format(**env)
         dst = dst_t.format(**env)
         if os.path.exists(src):
             if migrate_one(src, dst):
-                pass
+                migrated = True
     
     for name in config_migrations:
-        migrate_config(name, env)
+        if migrate_config(name, env):
+            migrated = True
     
     custom_src = custom_src_t.format(**env)
     custom_dst = custom_dst_t.format(**env)
     
     if os.path.exists(custom_src):
-        migrate_static_custom(custom_src, custom_dst)
+        if migrate_static_custom(custom_src, custom_dst):
+            migrated = True
+    
+    # write a marker to avoid re-running migration checks
+    ensure_dir_exists(env['jupyter_config'])
+    with open(os.path.join(env['jupyter_config'], 'migrated'), 'w') as f:
+        f.write(datetime.now().isoformat())
```

Comment:
It doesn't look like we're using this timestamp anywhere, but in case we later decide to, should it be `utcnow()`? If the user changes time zone between creating and using the timestamp, calculations based on local time will be off.